### ISSUE TITLE
1.20.0: Runtime panic in buildssa linter package

### ISSUE THREAD

**Issue by saschagrunert** [golangci/golangci-lint#800]
Hey :wave:, I'm encountering a runtime panic with the latest version. Here are the details:

1. Version: v1.20.0
2. Config file:
```yaml
---
run:
  build-tags:
    - apparmor
    - containers_image_ostree_stub
    - seccomp
    - selinux
    - test
    # containers/storage currently has btrfs included, but the images don't, so the linter fails.
    - exclude_graphdriver_btrfs
  concurrency: 6
  deadline: 5m
linters:
  disable-all: true
  enable:
    - bodyclose
    - deadcode
    - depguard
    - dupl
    - errcheck
    - gochecknoinits
    - goconst
    - gocritic
    - gocyclo
    - gofmt
    - goimports
    - golint
    - gosec
    - gosimple
    - govet
    - ineffassign
    - interfacer
    - maligned
    - misspell
    - nakedret
    - prealloc
    - scopelint
    - staticcheck
    - structcheck
    - stylecheck
    - typecheck
    - unconvert
    - unparam
    - unused
    - varcheck
    # - dogsled
    # - funlen
    # - gochecknoglobals
    # - gocognit
    # - godox
    # - lll
    # - whitespace
    # - wsl
linters-settings:
  errcheck:
    check-type-assertions: true
    check-blank: true
  gocritic:
    enabled-checks:
      # Diagnostic
      - appendAssign
      - argOrder
      - badCond
      - caseOrder
      - codegenComment
      - commentedOutCode
      - deprecatedComment
      - dupArg
      - dupBranchBody
      - dupCase
      - dupSubExpr
      - exitAfterDefer
      - flagDeref
      - flagName
      - nilValReturn
      - offBy1
      - sloppyReassign
      - weakCond
      # - octalLiteral

      # Performance
      - appendCombine
      - equalFold
      - hugeParam
      - indexAlloc
      - rangeExprCopy
      - rangeValCopy

      # Style
      - assignOp
      - boolExprSimplify
      - captLocal
      - commentFormatting
      - commentedOutImport
      - defaultCaseOrder
      - docStub
      - elseif
      - emptyFallthrough
      - emptyStringTest
      - hexLiteral
      - ifElseChain
      - methodExprCall
      - regexpMust
      - singleCaseSwitch
      - sloppyLen
      - stringXbytes
      - switchTrue
      - typeAssertChain
      - typeSwitchVar
      - underef
      - unlabelStmt
      - unlambda
      - unslice
      - valSwap
      - wrapperFunc
      - yodaStyleExpr

      # Opinionated
      - builtinShadow
      - importShadow
      - initClause
      - nestingReduce
      - paramTypeCombine
      - ptrToRefParam
      - typeUnparen
      - unnamedResult
      - unnecessaryBlock
  gocyclo:
    min-complexity: 122
  nakedret:
    max-func-lines: 15
```
3. Go environment: `go version && go env`
```> go version && go env
go version go1.12.10 linux/amd64
GOARCH="amd64"
GOBIN="/home/sascha/go/bin"
GOCACHE="/home/sascha/.cache/go-build"
GOEXE=""
GOFLAGS=""
GOHOSTARCH="amd64"
GOHOSTOS="linux"
GOOS="linux"
GOPATH="/home/sascha/go"
GOPROXY=""
GORACE=""
GOROOT="/usr/lib64/go/1.12"
GOTMPDIR=""
GOTOOLDIR="/usr/lib64/go/1.12/pkg/tool/linux_amd64"
GCCGO="gccgo"
CC="gcc"
CXX="g++"
CGO_ENABLED="1"
GOMOD=""
CGO_CFLAGS="-g -O2"
CGO_CPPFLAGS=""
CGO_CXXFLAGS="-g -O2"
CGO_FFLAGS="-g -O2"
CGO_LDFLAGS="-g -O2"
PKG_CONFIG="pkg-config"
GOGCCFLAGS="-fPIC -m64 -pthread -fmessage-length=0 -fdebug-prefix-map=/tmp/go-build136663385=/tmp/go-build -gno-record-gcc-switches"
>

```
4. Verbose output of running: `golangci-lint run -v`
```
> golangci-lint run -v
INFO [config_reader] Config search paths: [./ /home/sascha/go/src/github.com/cri-o/cri-o /home/sascha/go/src/github.com/cri-o /home/sascha/go/src/github.com /home/sascha/go/src /home/sascha/go /home/sascha /home /]
INFO [config_reader] Used config file .golangci.yml
INFO [lintersdb] Active 30 linters: [bodyclose deadcode depguard dupl errcheck gochecknoinits goconst gocritic gocyclo gofmt goimports golint gosec gosimple govet ineffassign interfacer maligned misspell nakedret prealloc scopelint staticcheck structcheck stylecheck typecheck unconvert unparam unused varcheck]
INFO [loader] Go packages loading at mode 575 (compiled_files|exports_file|types_sizes|deps|files|imports|name) took 903.387203ms
INFO [runner/filename_unadjuster] Pre-built 0 adjustments in 23.968534ms
WARN [linters context] Panic: buildssa: package "version" (isInitialPkg: true, needAnalyzeSource: true): interface conversion: *ssa.Call is not interface { ssa.setNum(int) }: missing method setNum: goroutine 7935 [running]:
runtime/debug.Stack(0xf7f9b9, 0x3c, 0xc001497590)
        /usr/lib64/go/1.12/src/runtime/debug/stack.go:24 +0x9d
github.com/golangci/golangci-lint/pkg/golinters/goanalysis.(*action).analyzeSafe.func1(0xc0011b58b0)
        /home/sascha/go/src/github.com/cri-o/cri-o/vendor/github.com/golangci/golangci-lint/pkg/golinters/goanalysis/runner.go:473 +0x1b5
panic(0xdd56a0, 0xc003bec4e0)
        /usr/lib64/go/1.12/src/runtime/panic.go:522 +0x1b5
honnef.co/go/tools/ssa.numberRegisters(0xc003beaf00)
        /home/sascha/go/src/github.com/cri-o/cri-o/vendor/honnef.co/go/tools/ssa/func.go:278 +0x114
honnef.co/go/tools/ssa.(*Function).finishBody(0xc003beaf00)
        /home/sascha/go/src/github.com/cri-o/cri-o/vendor/honnef.co/go/tools/ssa/func.go:410 +0x746
honnef.co/go/tools/ssa.(*builder).buildFunction(0xc004117a18, 0xc003beaf00)
        /home/sascha/go/src/github.com/cri-o/cri-o/vendor/honnef.co/go/tools/ssa/builder.go:2207 +0x316
honnef.co/go/tools/ssa.(*builder).buildFuncDecl(0xc004117a18, 0xc001cb5920, 0xc0021cc450)
        /home/sascha/go/src/github.com/cri-o/cri-o/vendor/honnef.co/go/tools/ssa/builder.go:2225 +0xed
honnef.co/go/tools/ssa.(*Package).build(0xc001cb5920)
        /home/sascha/go/src/github.com/cri-o/cri-o/vendor/honnef.co/go/tools/ssa/builder.go:2341 +0x88a
sync.(*Once).Do(0xc001cb594c, 0xc001855c80)
        /usr/lib64/go/1.12/src/sync/once.go:44 +0xb3
honnef.co/go/tools/ssa.(*Package).Build(0xc001cb5920)
        /home/sascha/go/src/github.com/cri-o/cri-o/vendor/honnef.co/go/tools/ssa/builder.go:2260 +0x54
honnef.co/go/tools/internal/passes/buildssa.run(0xc0000b0fa0, 0x4cb57e28, 0x1906e40, 0x2, 0x2)
        /home/sascha/go/src/github.com/cri-o/cri-o/vendor/honnef.co/go/tools/internal/passes/buildssa/buildssa.go:70 +0x27b
github.com/golangci/golangci-lint/pkg/golinters/goanalysis.(*action).analyze(0xc0011b58b0)
        /home/sascha/go/src/github.com/cri-o/cri-o/vendor/github.com/golangci/golangci-lint/pkg/golinters/goanalysis/runner.go:564 +0x9e8
github.com/golangci/golangci-lint/pkg/golinters/goanalysis.(*action).analyzeSafe(0xc0011b58b0)
        /home/sascha/go/src/github.com/cri-o/cri-o/vendor/github.com/golangci/golangci-lint/pkg/golinters/goanalysis/runner.go:476 +0x51
github.com/golangci/golangci-lint/pkg/golinters/goanalysis.(*loadingPackage).analyze.func3(0xc0007fe720, 0xc0011b58b0)
        /home/sascha/go/src/github.com/cri-o/cri-o/vendor/github.com/golangci/golangci-lint/pkg/golinters/goanalysis/runner.go:1030 +0x5f
created by github.com/golangci/golangci-lint/pkg/golinters/goanalysis.(*loadingPackage).analyze
        /home/sascha/go/src/github.com/cri-o/cri-o/vendor/github.com/golangci/golangci-lint/pkg/golinters/goanalysis/runner.go:1025 +0x2cf
WARN [linters context] Panic: buildssa: package "utils" (isInitialPkg: true, needAnalyzeSource: true): interface conversion: *ssa.Call is not interface { ssa.setNum(int) }: missing method setNum: goroutine 22764 [running]:
runtime/debug.Stack(0xf7f9b9, 0x3c, 0xc004f9d248)
        /usr/lib64/go/1.12/src/runtime/debug/stack.go:24 +0x9d
github.com/golangci/golangci-lint/pkg/golinters/goanalysis.(*action).analyzeSafe.func1(0xc00d213c90)
        /home/sascha/go/src/github.com/cri-o/cri-o/vendor/github.com/golangci/golangci-lint/pkg/golinters/goanalysis/runner.go:473 +0x1b5
panic(0xdd56a0, 0xc016079f50)
        /usr/lib64/go/1.12/src/runtime/panic.go:522 +0x1b5
honnef.co/go/tools/ssa.numberRegisters(0xc028023540)
        /home/sascha/go/src/github.com/cri-o/cri-o/vendor/honnef.co/go/tools/ssa/func.go:278 +0x114
honnef.co/go/tools/ssa.(*Function).finishBody(0xc028023540)
        /home/sascha/go/src/github.com/cri-o/cri-o/vendor/honnef.co/go/tools/ssa/func.go:410 +0x746
honnef.co/go/tools/ssa.makeWrapper(0xc018eebd60, 0xc02706e9b0, 0x0)
        /home/sascha/go/src/github.com/cri-o/cri-o/vendor/honnef.co/go/tools/ssa/wrappers.go:135 +0x70a
honnef.co/go/tools/ssa.(*Program).addMethod(0xc018eebd60, 0xc027d2b7f0, 0xc02706e9b0, 0xc027d2b7f0)
        /home/sascha/go/src/github.com/cri-o/cri-o/vendor/honnef.co/go/tools/ssa/methods.go:84 +0x13f
honnef.co/go/tools/ssa.(*Program).needMethods(0xc018eebd60, 0x10d8220, 0xc027d2b7e0, 0xc028022b00)
        /home/sascha/go/src/github.com/cri-o/cri-o/vendor/honnef.co/go/tools/ssa/methods.go:173 +0x853
honnef.co/go/tools/ssa.(*Program).needMethods(0xc018eebd60, 0x10d81a0, 0xc0052e6930, 0x0)
        /home/sascha/go/src/github.com/cri-o/cri-o/vendor/honnef.co/go/tools/ssa/methods.go:215 +0x72c
honnef.co/go/tools/ssa.(*Program).needMethodsOf(0xc018eebd60, 0x10d81a0, 0xc0052e6930)
        /home/sascha/go/src/github.com/cri-o/cri-o/vendor/honnef.co/go/tools/ssa/methods.go:145 +0x5d
honnef.co/go/tools/ssa.(*Package).build(0xc027c7ecc0)
        /home/sascha/go/src/github.com/cri-o/cri-o/vendor/honnef.co/go/tools/ssa/builder.go:2272 +0x110
sync.(*Once).Do(0xc027c7ecec, 0xc00752cc80)
        /usr/lib64/go/1.12/src/sync/once.go:44 +0xb3
honnef.co/go/tools/ssa.(*Package).Build(0xc027c7ecc0)
        /home/sascha/go/src/github.com/cri-o/cri-o/vendor/honnef.co/go/tools/ssa/builder.go:2260 +0x54
honnef.co/go/tools/internal/passes/buildssa.run(0xc018eebcc0, 0x142ee6fb9, 0x1906e40, 0x2, 0x2)
        /home/sascha/go/src/github.com/cri-o/cri-o/vendor/honnef.co/go/tools/internal/passes/buildssa/buildssa.go:70 +0x27b
github.com/golangci/golangci-lint/pkg/golinters/goanalysis.(*action).analyze(0xc00d213c90)
        /home/sascha/go/src/github.com/cri-o/cri-o/vendor/github.com/golangci/golangci-lint/pkg/golinters/goanalysis/runner.go:564 +0x9e8
github.com/golangci/golangci-lint/pkg/golinters/goanalysis.(*action).analyzeSafe(0xc00d213c90)
        /home/sascha/go/src/github.com/cri-o/cri-o/vendor/github.com/golangci/golangci-lint/pkg/golinters/goanalysis/runner.go:476 +0x51
github.com/golangci/golangci-lint/pkg/golinters/goanalysis.(*loadingPackage).analyze.func3(0xc0056d4ca0, 0xc00d213c90)
        /home/sascha/go/src/github.com/cri-o/cri-o/vendor/github.com/golangci/golangci-lint/pkg/golinters/goanalysis/runner.go:1030 +0x5f
created by github.com/golangci/golangci-lint/pkg/golinters/goanalysis.(*loadingPackage).analyze
        /home/sascha/go/src/github.com/cri-o/cri-o/vendor/github.com/golangci/golangci-lint/pkg/golinters/goanalysis/runner.go:1025 +0x2cf
INFO [runner] processing took 2.142µs with stages: max_same_issues: 503ns, skip_dirs: 267ns, nolint: 193ns, max_from_linter: 166ns, cgo: 118ns, path_prettifier: 115ns, diff: 111ns, filename_unadjuster: 108ns, autogenerated_exclude: 107ns, identifier_marker: 104ns, skip_files: 100ns, exclude-rules: 47ns, source_code: 42ns, path_shortener: 41ns, max_per_file_from_linter: 40ns, exclude: 40ns, uniq_by_line: 40ns
INFO [runner] linters took 11.265939089s with stages: goanalysis_metalinter: 10.024545752s, unused: 1.241355175s
INFO File cache stats: 196 entries of total size 893.2KiB
INFO Memory: 123 samples, avg is 732.8MB, max is 1146.9MB
INFO Execution took 12.201709897s
```

**Comment by jirfag** [comment]
Please, use binary release. It’s the only stable way to run any tool. Your panic is because of old Go and Go compiler bug
